### ISSUE TITLE
[java] Exception applying rule ConfusingArgumentToVarargsMethod

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#5027]
**Affects PMD Version:** 7.1.0

**Description:**

JUnitAssertionsShouldIncludeMessage throws an exception

**Exception Stacktrace:**

```
org.apache.commons.lang3.exception.ContextedRuntimeException: java.lang.IllegalArgumentException: Bad intersection, unrelated class types java.lang.Short and com.fasterxml.jackson.databind.JsonNode in [com.fasterxml.jackson.databind.JsonNode, java.lang.Short]
Exception Context:
	[1:Resolving type of=!debug only! [MethodCall:84:4]elements.add(body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), b(truncated)]
	[2:Rule applied on node=!debug only! [ArgumentList:84:16](body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), body.arrayNod(truncated)]
---------------------------------
    at net.sourceforge.pmd.util.AssertionUtil.contexted (AssertionUtil.java:236)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaTypeNode.getTypeMirror (AbstractJavaTypeNode.java:54)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.getTypeMirror (ASTMethodCall.java:22)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaTypeNode.getTypeMirror (AbstractJavaTypeNode.java:39)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.getTypeMirror (ASTMethodCall.java:22)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaTypeNode.forceTypeResolution (AbstractJavaTypeNode.java:29)
    at net.sourceforge.pmd.lang.java.ast.AbstractInvocationExpr.getOverloadSelectionInfo (AbstractInvocationExpr.java:27)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.getOverloadSelectionInfo (ASTMethodCall.java:22)
    at net.sourceforge.pmd.lang.java.rule.errorprone.ConfusingArgumentToVarargsMethodRule.visit (ConfusingArgumentToVarargsMethodRule.java:33)
    at net.sourceforge.pmd.lang.java.ast.ASTArgumentList.acceptVisitor (ASTArgumentList.java:27)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:30)
    at net.sourceforge.pmd.lang.rule.RuleReference.apply (RuleReference.java:415)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:57)
    at net.sourceforge.pmd.lang.rule.internal.RuleSets.apply (RuleSets.java:155)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1570)
Caused by: java.lang.IllegalArgumentException: Bad intersection, unrelated class types java.lang.Short and com.fasterxml.jackson.databind.JsonNode in [com.fasterxml.jackson.databind.JsonNode, java.lang.Short]
    at net.sourceforge.pmd.lang.java.types.Lub.glb (Lub.java:344)
    at net.sourceforge.pmd.lang.java.types.TypeSystem.glb (TypeSystem.java:723)
    at net.sourceforge.pmd.lang.java.types.internal.infer.ReductionStep$3.solve (ReductionStep.java:52)
    at net.sourceforge.pmd.lang.java.types.internal.infer.InferenceContext.solveBatchProgressed (InferenceContext.java:479)
    at net.sourceforge.pmd.lang.java.types.internal.infer.InferenceContext.solve (InferenceContext.java:459)
    at net.sourceforge.pmd.lang.java.types.internal.infer.InferenceContext.solve (InferenceContext.java:435)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.tryToSolve (Infer.java:620)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateImpl (Infer.java:572)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateMaybeNoInfer (Infer.java:545)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateMethod (Infer.java:389)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateMethodOrCtor (Infer.java:369)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.logInference (Infer.java:352)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.finishInstantiation (Infer.java:332)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.determineInvocationTypeOrFail (Infer.java:248)
    at net.sourceforge.pmd.lang.java.types.internal.infer.ExprCheckHelper.isInvocationCompatible (ExprCheckHelper.java:166)
    at net.sourceforge.pmd.lang.java.types.internal.infer.ExprCheckHelper.isCompatible (ExprCheckHelper.java:155)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.addBoundOrDefer (Infer.java:914)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.addArgsConstraints (Infer.java:872)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateMaybeNoInfer (Infer.java:505)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateMethod (Infer.java:389)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.instantiateMethodOrCtor (Infer.java:369)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.logInference (Infer.java:352)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.computeCompileTimeDecl (Infer.java:292)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.getCompileTimeDecl (Infer.java:254)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.goToInvocationWithFallback (Infer.java:187)
    at net.sourceforge.pmd.lang.java.types.internal.infer.Infer.inferInvocationRecursively (Infer.java:176)
    at net.sourceforge.pmd.lang.java.types.ast.internal.PolyResolution.inferInvocation (PolyResolution.java:254)
    at net.sourceforge.pmd.lang.java.types.ast.internal.PolyResolution.polyTypeOtherCtx (PolyResolution.java:126)
    at net.sourceforge.pmd.lang.java.types.ast.internal.PolyResolution.computePolyType (PolyResolution.java:116)
    at net.sourceforge.pmd.lang.java.types.ast.internal.LazyTypeResolver.handlePoly (LazyTypeResolver.java:267)
    at net.sourceforge.pmd.lang.java.types.ast.internal.LazyTypeResolver.visit (LazyTypeResolver.java:272)
    at net.sourceforge.pmd.lang.java.types.ast.internal.LazyTypeResolver.visit (LazyTypeResolver.java:95)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.acceptVisitor (ASTMethodCall.java:75)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaTypeNode.getTypeMirror (AbstractJavaTypeNode.java:51)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.getTypeMirror (ASTMethodCall.java:22)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaTypeNode.getTypeMirror (AbstractJavaTypeNode.java:39)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.getTypeMirror (ASTMethodCall.java:22)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaTypeNode.forceTypeResolution (AbstractJavaTypeNode.java:29)
    at net.sourceforge.pmd.lang.java.ast.AbstractInvocationExpr.getOverloadSelectionInfo (AbstractInvocationExpr.java:27)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodCall.getOverloadSelectionInfo (ASTMethodCall.java:22)
    at net.sourceforge.pmd.lang.java.rule.errorprone.ConfusingArgumentToVarargsMethodRule.visit (ConfusingArgumentToVarargsMethodRule.java:33)
    at net.sourceforge.pmd.lang.java.ast.ASTArgumentList.acceptVisitor (ASTArgumentList.java:27)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:30)
    at net.sourceforge.pmd.lang.rule.RuleReference.apply (RuleReference.java:415)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:57)
    at net.sourceforge.pmd.lang.rule.internal.RuleSets.apply (RuleSets.java:155)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1570)
[WARNING] Exception occurred on node !debug only! [ArgumentList:84:16](body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), body.arrayNod(truncated)
[WARNING] There are 2 PMD processing errors:
[WARNING] C:\Users\Gili\Documents\licensed-app\server\src\main\java\app\licensed\server\resource\api\version1\organization\OrganizationsResource.java: ContextedRuntimeException: java.lang.IllegalArgumentException: Bad intersection, unrelated class types java.lang.Short and com.fasterxml.jackson.databind.JsonNode in [com.fasterxml.jackson.databind.JsonNode, java.lang.Short]
Exception Context:
	[1:Resolving type of=!debug only! [MethodCall:84:4]elements.add(body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), b(truncated)]
	[2:Rule applied on node=!debug only! [MethodCall:84:4]elements.add(body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), b(truncated)]
---------------------------------
C:\Users\Gili\Documents\licensed-app\server\src\main\java\app\licensed\server\resource\api\version1\organization\OrganizationsResource.java: ContextedRuntimeException: java.lang.IllegalArgumentException: Bad intersection, unrelated class types java.lang.Short and com.fasterxml.jackson.databind.JsonNode in [com.fasterxml.jackson.databind.JsonNode, java.lang.Short]
Exception Context:
	[1:Resolving type of=!debug only! [MethodCall:84:4]elements.add(body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), b(truncated)]
	[2:Rule applied on node=!debug only! [ArgumentList:84:16](body.objectNode().
				put(MetadataProperty.LINK.toString(), uri.toString()).
				set(MetadataProperty.RELATIONS.toString(), body.arrayNod(truncated)]
---------------------------------
```

**Code Sample demonstrating the issue:**

The exception is triggered by this code:

```java
ObjectNode body = scope.getObjectMapper().createObjectNode();

ArrayNode elements = body.arrayNode();
for (OrganizationRow row : organizationRows)
{
	URI uri = router.getUri(new OrganizationResource(row.getId().getValue()));
	elements.add(body.objectNode().
		put(MetadataProperty.LINK.toString(), uri.toString()).
		set(MetadataProperty.RELATIONS.toString(), body.arrayNode().
			add(Relation.ELEMENT.toString())));
}
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

`mvn clean install`

**Running PMD through:** maven-pmd-plugin 3.22.0
JDK 22.0.1

**Comment by oowekyala** [comment]
Might have been fixed by #4994

**Comment by cowwoc** [comment]
Confirmed. This seems to be fixed in version 7.2.0-SNAPSHOT. Thanks for the head's up.

**Comment by jsotuyod** [comment]
Closing as duplicate then. Thank you for checking!
